### DISCUSSION TITLE
Provide a flag when creating a new Analog app to default to config based routing

### DISCUSSION THREAD

**Discussion by tayambamwanza** [analogjs/analog#1367] [Ideas]
### Which scope/s are relevant/related to the feature request?

create-analog

### Information

This request is more of a personal preference, I've used file based routing by working on a svelte project, it's a good system but personally I do prefer config based routing, would it be possible to include a flag/arg when creating a new project to stick to Angular's default routing?

I would also like to state that my desire for config based routing here is not a reflection on the work done on file based routing in Analog. I do genuanly appreciate the work that's been put into it and I know a lot of good work has gone into it and would encourage anyone to at least try  it but it's just not my preference.

### Describe any alternatives/workarounds you're currently using

Undoing the file based routing setup and reconfiguring default Angular setup.

### I would be willing to submit a PR to fix this issue

- [X] Yes
- [ ] No

**Comment by brandonroberts** [comment]
If you want to manually configure routes, I think config based routing is easy enough to switch out. We could add a section in the docs on how to do this

**Comment by tayambamwanza** [comment]
@brandonroberts Thanks for giving this consideration and opening it up to discussion.

This is definitely a "convenience" request, to give context on what prompted me to request this feature:

I wrote a personal guide for setting up apps in an NX workspace.

The initial app creation is very simple because it's practically one line that I can copy/paste and viola, project is setup the way I need it. I am also now mentoring others to get clients/jobs and training them to use my setup (community outreach not for profit).

In order to include Analog as part of the guide I now have to write additional instructions just to get started (scss and config routing) which used to be a one liner. This is what prompted me to make the request.

I would really like to keep that "one line" experience for project setup and I'm even willing to do the work to implement it.

I know this isn't a super critical feature and of course it does mean excluding a feature that is advertised for Analog and for me it's purely for convenience but another possible benefit I can think of is maybe to some Angular Devs as good middle ground to migrate gradually by opting into a more familiar environment initially and adopting more Analog features as they get more experience with it.

**Reply by brandonroberts** [reply]
Understandable. I think its reasonable to use the Angular Router as-is, and it is a supported use case. 

My reservation is that if it becomes a default option, then people will think that some of the other features, like content routes, server-side data fetching, route metadata, and others still work, but they are built around the filesystem router so there may be some confusion there. 

Another option would be to create your own template repository that's suited to your preferences.

**Reply by tayambamwanza** [reply]
@brandonroberts Ah ok, so there are some features that depend on it, ok in that case maybe just some documentation and what the tradoffs are will do, as well as what are the limitations of current router. Perhaps I'll come back to this and add to the documentation and perhaps I'll just make my own seperate migration tool so it's unofficial and not recommended.

**Reply by tayambamwanza** [reply]
@brandonroberts so would that question I asked you about using something like nodemailer break with normal router?

**Reply by brandonroberts** [reply]
No, it wouldn't. That still goes through API routes on the server, and not through the router

**Reply by tayambamwanza** [reply]
@brandonroberts understood, and you have to use one or the other right? No hybrid approach available?

**Reply by tayambamwanza** [reply]
@brandonroberts I think I'll stick to the file based routing for now and give feedback if I encounter any issues in the conversion. If you could please keep this open and I will use it to note when I run into issues, just for a big picture on what migration experience is like, if you don't mind.

**Reply by brandonroberts** [reply]
A hybrid approach also works. If you have some pre-configured routes, they get picked up also. All routes will be merged into a single array.

```ts
providers: [
  provideRouter(routes),
  provideFileRouter()
]
```

**Reply by tayambamwanza** [reply]
@brandonroberts Ah sorry I responded already, I'll try hybrid first and if it doesn't work just stick to file based, either way I'll give feedback.

**Reply by brandonroberts** [reply]
👍
